### DISCUSSION TITLE
One-click version upgrade button

### DISCUSSION THREAD

**Discussion by vladjohnson** [outline/outline#3415] [Ideas]
Dear Outline Community,

As a frequent user of your product, I would like to suggest a helpful feature that I think would make Outline even better: on-click version upgrade button. As we all know, to upgrade Outline currently, you need to upgrade container and migrate the database. For many people this is not convenient and I think that having a single button in the settings that upgrades version to the latest would be simply amazing. This feature is already implemented in many products, i.e. Organizr. I understand that it is pretty challenging to implement, but I hope someone can introduce this improvement to one of the future releases. Thank you for your hard work and way to go, Team Outline!

**Comment by almereyda** [comment]
A similar feature is available at Discourse. Yet they rely on a custom Docker scheduling app, called `launcher`, to orchestrate their sets of services.

I am not sure if it is the responsibility of the software vendor to provide resource orchestration and process scheduling functionality to their end-user applications.

How do you consider upgrading Outline inconveninent? It is no different than installing it, so that step has already been made. And the Outline Docker container can be configured to run the migrations automatically, if desired:

```yaml
  outline:
    image: outlinewiki/outline:0.62.0
    command: sh -c "yarn sequelize:migrate --env production-ssl-disabled && yarn start"
```

Then upgrading Outline turns into updating the version tag of the application, and rebuilding the container.

This is a common operation, and should be supported by your container hosting option.

With immutable, stateless container images, software is not installed *in place* on a *server*, as it used to be, where the state of source code trees and system units could be updated, but *instantiated* from an immutable source and eventually replaced in a *container environment*.

It is considered bad practice to rebuild the state and thus source of a container, which leads us to the solution provided by Discourse, where containers are rebuilt by the `launcher` through the web app, which thus has root access to the whole system. Is it something like that, which you would like to achieve?

To me it sounds a little incompatible to and impractical to implement with current (container) hosting environments.
